iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1011,25 +1018,31 @@ public URI getURIForAuth(org.apache.hadoop.hive.metastore.api.Table hmsTable) th
     Optional<String> metadataLocation =
         SessionStateUtil.getProperty(conf, BaseMetastoreTableOperations.METADATA_LOCATION_PROP);
     if (metadataLocation.isPresent()) {
-      authURI.append(encodeString(metadataLocation.get()));
+      authURI.append(encodeString(getLocation(metadataLocation.get(), getDefaultCreateLocation(), conf)));
```

Comment:
could you please refactor `getLocation(....)` to `getPathForAuth(String path)`
````
return encodeString(location)
````